glossary.ftd

```diff
@@ -1,22 +1,219 @@
 -- ft.page: Glossary
 
-For readers/writers of document by id, we have three kinds of id: `id: /foo/`,
-`full-id: /foo/-/x/`, `filename: foo.ftd`.
+An FPM Glossary
 
 
-Since we need to get `id`, `full-id` and `filename` of current document, to power
-document-readers, lets create processors `document-id`, `document-full-id` and
-`document-filename` as helpers as well. These three processors will return ids
-for current document.
+
+-- ft.h2: client project
+
+An [fpm project](glossary/#fpm-project) which is a clone to the
+[server project](glossary/#server-project) and tracks it. The interaction
+with server project is established using [sync](glossary/#sync).
+
+
+-- ft.h2: clone
+
+Clone a [server project](glossary/#server-project) into a new directory
+which becomes a [client](glossary/#server-project).
+
+
+-- ft.h2: conflict-state
+
+The file is said to be in the `conflict-state` if the file gets in conflict with
+server latest [version](glossary/#version) of the same file.
+
+```
+The file, in conflict-state, is never considered to be synced unless its
+conflict is resolved.
+```
+
+
+-- ft.h2: document
+
+The [file](glossary/#file) with extension that are compiled using FPM.
+Currently, there are two such extensions `.md` and `.ftd`.
+
+
+-- ft.h2: document-name
+
+The unique identifier of a [document](glossary/#document).
+
+This can accessed using [`document-filename`](processors/document-filename/)
+processor.
```

Comment:
This does not belong to this file. We do not mention processors. We are only defining terms. `document-name` and `id` or `document-id` conflict. You have assumed `document-id` is not a term, but is the name of processor alone. We should call the processor the `document-id processor`.

Reply:
Ideally processor names etc are not terms. Terms are a bit more abstract.